Wasm/Engine.lean

```diff
@@ -180,139 +165,152 @@ def findLocalByName? (ls : List (Option String × Option StackEntry))
   | y :: [] => y.2
   | _ => .none
 
+/-
+The `List Operation` in the `Continuation` type represents an optional 'final'
+sequence of instructions – which should replace the rest of the instructions
+in the currently executed sequence, which in turn emulates the continuation
+jump, except it's more like 'continuation unwinding'.
+
+Semantics:
+- `.none` means there is no continuation, like in most simple/data instructions.
+- `some []` means there is a continuation, but it is empty, ending the block.
+- `some ops` means "drop the rest of whatever you're doing and run this instead"
+
+-------------/
+abbrev Continuation := List Operation
+
+/-
+This is a readability helper monad stack abbreviation for use in handling
+flow correctly when executing branch instructions like `br` and `br_if`,
+as well as just for easy handling of the stack and possible errors.
+
+The monads in the stack:
+- `Except EngineErrors` throws _real_ execution errors. It's on the very bottom
+  of the stack because we don't care about anything else, like recovery, if
+  there's an engine error.
+- `StateT (List StackEntry)` carries the stack around for ease of handling.
+- `ExceptT Continuation` doesn't handle real exceptions, instead it serves as
+  a way to throw continuations through the execution cycle like described above.
+  It's the outermost transformer because when a continuation throw does occur,
+  we want both the `Continuation` and the `List StackEntry` that comes with it
+  from the `StateT` layer.
+
+-/
+abbrev EngineM :=
+  ExceptT Continuation (StateT (List StackEntry) (Except EngineErrors))
+
+instance : Inhabited (EngineM α) where
+  default := throw default
+
+def throwEE : EngineErrors → EngineM α := ExceptT.lift ∘ throw
+def raiseCont : List Operation → EngineM α := throw
+
+def push : StackEntry → EngineM PUnit := fun x => do set $ x :: (←get)
+def pile : List StackEntry → EngineM PUnit := fun xs => do set $ xs ++ (←get)
+def σmap : (List StackEntry → List StackEntry) → EngineM PUnit :=
+  fun f => do set $ f (←get)
+def bite : EngineM StackEntry := do match (←get) with
```

Comment:
LGTM, maybe it could be straight after `push`?. Like push-bite / pile-\sigmamap?